### DISCUSSION TITLE
What is the `placement` attribute of `<non-arpeggiate>` good for?

### DISCUSSION THREAD

**Discussion by lemzwerg** [w3c-cg/musicxml#540] [General]
As the title says.  Looking into `musicxml.xsd` I see that the definition of `<non-arpeggiate>` explicitly contains

```xml
<xs:attributeGroup ref="placement"/>
```

so it seems that its inclusion is intentional.  However, putting an arpeggio bracket 'below' or 'above' a chord makes no sense to me.  What am I missing?

**Comment by mscuthbert** [comment]
I don't think you're missing anything and your observation that it doesn't seem to do anything makes sense to me.  My initial thought was that it probably came from a grouped entity that included placement, but I don't see that in the history.  It has definitely been around since at least v.1.0.

What if 4.1 notes that placement does not have any known effect for `<non-arpeggiate>` and may become deprecated in a future release.

Another thing that I would like to make explicit is which of the top/bottom the color, position, etc. of this goes (top? bottom? first?) and  whether the top and bottom should use the same optional-unique-id or different ones.

**Reply by lemzwerg** [reply]
> What if 4.1 notes that placement does not have any known effect for <non-arpeggiate> and may become deprecated in a future release.

Sounds good.  Dito for your suggestions to improve the documentation.

**Reply by lemzwerg** [reply]
BTW, I've filed issue #541 to track this – also covering `<arpeggiate>`, which has the same issue.

**Reply by mdgood** [reply]
This sounds good. I have no recollection of why this was there. I'm guessing that it was my (mis)interpretation of what was possible in MuseData.

**Comment by lemzwerg** [comment]
I have another question: Does an arpeggio always span the whole chord?  The documentation is not very clear on this IMHO.  I have never seen a chord where the wavy line doesn't cover all notes of it (contrary to arpeggio brackets), but this is just me...

**Reply by mdgood** [reply]
No, you need to use an `<arpeggiate/>` element on each note of the chord. Arpeggios sometimes only cover part of a chord, usually the lower notes of a chord, though I don't have an example to share at the moment.

**Reply by lemzwerg** [reply]
> No, you need to use an `<arpeggiate/>` element on each note of the chord.

I'm confused now.  Your 'no' is an answer to my question, I guess, but then you say that `<arpeggiate>` needs to be on each note of the chord, which sounds like a contradiction.

If I understand you correctly, the following is possible, right (using pseudo-MusicXML for brevity)?

```xml
<note>
  <pitch><step>C</step><octave>4</octave></pitch>
  <notations><arpeggiate/></notations>
</note>
<note>
  <chord/>
  <pitch><step>E</step><octave>4</octave></pitch>
  <notations><arpeggiate/></notations>
</note>
<note>
  <chord/>
  <pitch><step>G</step><octave>4</octave></pitch>
  <notations><arpeggiate/></notations>
</note>
<note>
  <pitch><step>C</step><octave>5</octave></pitch>
</note>
<note>
  <chord/>
  <pitch><step>E</step><octave>5</octave></pitch>
</note>
<note>
  <chord/>
  <pitch><step>G</step><octave>5</octave></pitch>
</note>
```

![image](https://github.com/user-attachments/assets/c7b6d048-012f-4896-b116-4dbcc3b13832)

**Reply by mdgood** [reply]
That's correct. I meant you need to use the `<arpeggiate/>` element on each note of the chord that's part of the arpeggio.

**Reply by lemzwerg** [reply]
OK, thanks for the clarification.

**Comment by lemzwerg** [comment]
Aaand another question.  Consider nested arpeggio brackets like the following (this might happen as an editorial advice to help resolve, say, a chord in Baroque violin music under special circumstances).

![image](https://github.com/user-attachments/assets/2da854bd-9558-4a39-a6de-8df69a3bb115)

I suppose it would be straightforward to use the `number` attribute of `<non-arpeggiate>` to implement that, right?  If so I suggest to extend the description of `number` to cover multiple arpeggio brackets for a single chord.
